plugins/dominant-color-images/hooks.php

```diff
@@ -65,16 +65,17 @@ function dominant_color_update_attachment_image_attributes( $attr, WP_Post $atta
 		$attr['data-has-transparency'] = $image_meta['has_transparency'] ? 'true' : 'false';
 
 		$class = $image_meta['has_transparency'] ? 'has-transparency' : 'not-transparent';
-		if ( empty( $attr['class'] ) ) {
-			$attr['class'] = $class;
-		} else {
+
+		if ( isset( $attr['class'] ) && is_string( $attr['class'] ) && '' !== $attr['class'] ) {
 			$attr['class'] .= ' ' . $class;
+		} else {
+			$attr['class'] = $class;
 		}
 	}
 
-	if ( ! empty( $image_meta['dominant_color'] ) ) {
+	if ( isset( $image_meta['dominant_color'] ) && is_string( $image_meta['dominant_color'] ) && '' !== $image_meta['dominant_color'] ) {
 		$attr['data-dominant-color'] = esc_attr( $image_meta['dominant_color'] );
-		$style_attribute             = empty( $attr['style'] ) ? '' : $attr['style'];
+		$style_attribute             = isset( $attr['style'] ) && is_string( $attr['style'] ) ? $attr['style'] : '';
```

Comment:
Nit-pick, we could also make this:
```suggestion
		$style_attribute             = isset( $attr['style'] ) ? (string) $attr['style'] : '';
```

Works too, maybe a bit more forgiving.

Reply:
But it could result in a PHP warning: https://3v4l.org/NDQt8

Reply:
Okay, that's fair.